src/line_breaks.rs

```diff
@@ -0,0 +1,367 @@
+//! Line breaking functionality.
+
+#[cfg(feature = "unicode-linebreak")]
+use crate::core::skip_ansi_escape_sequence;
+use crate::core::Word;
+
+/// Describes where a line break can occur.
+///
+/// The simplest approach is say that a line can end after one or more
+/// ASCII spaces (`' '`). This works for Western languages without
+/// emojis. A more complex approach is to use the Unicode line
+/// breaking algorithm, which finds break points in non-ASCII text.
+///
+/// The line breaks occur between words, please see the
+/// [`WordSplitter`](crate::WordSplitter) trait for options of how
+/// to handle hyphenation of individual words.
+pub trait LineBreakAlgorithm: std::clone::Clone {
+    /// Find line break opportunities in `line` and return these as words.
+    fn break_line<'a>(&self, line: &'a str) -> Box<dyn Iterator<Item = Word<'a>> + 'a>;
+}
+
+/// Default line breaking algorithm when the `unicode-linebreak` Cargo
+/// feature is enabled.
+#[cfg(feature = "unicode-linebreak")]
+pub type DefaultLineBreakAlgorithm = UnicodeLineBreaks;
```

Comment:
This huge identifier is annoying to see and it creates an annoying indirection... it's there since I feel the unicode-linebreak crate is an optional dependency, yet I would like to use it by default if it is available. So to avoid tons of `#[cfg(feature = "unicode-linebreak")]` in the rest of the code, I define this type here.

Reply:
It turns out I can use a macro for this! This allows me to keep the shenanigans here internal to the library.